leshan-tl-javacoap-server/src/main/java/org/eclipse/leshan/transport/javacoap/endpoint/JavaCoapServerEndpointsProvider.java

```diff
@@ -0,0 +1,114 @@
+import org.eclipse.leshan.transport.javacoap.resource.RegistrationResource;
+import org.eclipse.leshan.transport.javacoap.resource.ResourcesService;
+import org.eclipse.leshan.transport.javacoap.resource.SendResource;
+
+import com.mbed.coap.server.CoapServer;
+
+public class JavaCoapServerEndpointsProvider implements LwM2mServerEndpointsProvider {
+
+    private CoapServer coapServer;
+    private final int coapPort;
+
+    private JavaCoapServerEndpoint lwm2mEndpoint;
+
+    public JavaCoapServerEndpointsProvider(int coapPort) {
+        this.coapPort = coapPort;
+    }
+
+    @Override
+    public void createEndpoints(UplinkRequestReceiver requestReceiver, LwM2mNotificationReceiver notificationReceiver,
+            ServerEndpointToolbox toolbox, ServerSecurityInfo serverSecurityInfo, LeshanServer server) {
+
+        // TODO we should get endpoint URI dynamically in Resources
+        URI endpointURI = EndpointUriUtil.createUri("coap", "0.0.0.0", coapPort);
+
+        // create Resources / Routes
+        RegistrationResource registerResource = new RegistrationResource(requestReceiver, toolbox.getLinkParser(),
+                endpointURI);
+        ResourcesService resources = ResourcesService.builder() //
+                .add("/rd/*", registerResource) //
+                .add("/rd", registerResource)//
+                .add("/dp",
+                        new SendResource(requestReceiver, toolbox.getDecoder(), toolbox.getProfileProvider(),
+                                endpointURI))//
+                .build();
+        coapServer = CoapServer.builder().transport(coapPort).route(resources).build();
+
+        lwm2mEndpoint = new JavaCoapServerEndpoint(endpointURI, coapServer, new ServerCoapMessageTranslator(), toolbox,
+                notificationReceiver, server.getRegistrationStore());
+    }
+
+    @Override
+    public List<LwM2mServerEndpoint> getEndpoints() {
+        // We support only one endpoint for now
+        // TODO I don't know if 1 java-coap server can support several endpoints ?
+        if (lwm2mEndpoint == null) {
+            return Collections.emptyList();
+        } else {
+            return Arrays.asList(lwm2mEndpoint);
+        }
+    }
+
+    @Override
+    public LwM2mServerEndpoint getEndpoint(URI uri) {
+        // We support only one endpoint for now
+        // TODO I don't know if 1 java-coap server can support several endpoints ?
+        if (lwm2mEndpoint != null && lwm2mEndpoint.getURI().equals(uri))
+            return lwm2mEndpoint;
+        else
+            return null;
+    }
+
+    @Override
+    public void start() {
+        try {
+            coapServer.start();
+        } catch (IllegalStateException | IOException e) {
+            // TODO handle this correctly
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void stop() {
+        // TODO in Leshan stop means "we can restart after a stop", so we should check what means stop for java-coap
```

Comment:
There is no restart after stop, need to create new instance to start again. 
Out of curiosity, what's the use case for restart?

Reply:
To be honest I'm not so sure if this is really needed, I know that we use it in some tests (especially at LWM2M client side to simulate that device is sleeping).

Californium has a start / stop / destroy mechanism where stop means that's you can restart unlike destroy.
Stop is kind of "pause". 

As Leshan was strongly tied to californium, we follow this API.
Then when I did first transport abstraction layer, I keep it... 
Always easier to add feature than remove it, but maybe better to just keep start() and destroy(). :thinking: